### PR TITLE
SSCS-7580 Allowed Scenario : Setting schedule3 descriptors on payload, and adding tests

### PR DESCRIPTION
**Pull request by michaellavelle**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-7580

### Change description ###

Setting schedule3 descriptors on payload

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionService.java [hunk 1]
```diff
@@ -305,45 +305,55 @@ protected void setPipDescriptorsAndPoints(WriteFinalDecisionTemplateBodyBuilder
...
         EsaDecisionNoticeQuestionService questionService = (EsaDecisionNoticeQuestionService)decisionNoticeService.getQuestionService("ESA");
-        return getDescriptorsFromQuestionKeys("ESA", key -> questionService.extractQuestionFromKey(EsaActivityQuestionKey.getByKey(key)), caseData, questionKeys);
+        return getQuestionAndAnswerDescriptorsFromQuestionKeys("ESA", key -> questionService.extractQuestionFromKey(EsaActivityQuestionKey.getByKey(key)), caseData, questionKeys);
     }
 
-    protected List<Descriptor> getDescriptorsFromQuestionKeys(String benefitType, ActivityQuestionLookup activityQuestionlookup, SscsCaseData caseData, List<String> questionKeys) {
+    protected List<Descriptor> getEsaSchedule3DescriptorsFromQuestionKeys(SscsCaseData caseData, List<String> questionKeys) {
+        EsaDecisionNoticeQuestionService questionService = (EsaDecisionNoticeQuestionService)decisionNoticeService.getQuestionService("ESA");
+        return getQuestionOnlyDescriptorsFromQuestionKeys("ESA", key -> questionService.extractQuestionFromKey(EsaSchedule3QuestionKey.getByKey(key)), caseData, questionKeys);
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionService.java:353]: There are lot's a 'ESA' strings, perhaps use a constant.

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java [hunk 2]
```diff
@@ -865,4 +865,164 @@ public void givenSupportGroupRefusedScenario_Regulation29FieldIsNotPopulatedAndP
...
+        sscsCaseData.setWriteFinalDecisionGenerateNotice("yes");
+
+        sscsCaseData.setEsaWriteFinalDecisionPhysicalDisabilitiesQuestion(Arrays.asList("mobilisingUnaided"));
+
+        // 0 points - high, which means regulation 29 is not applicable
+        sscsCaseData.setEsaWriteFinalDecisionMobilisingUnaidedQuestion("mobilisingUnaided1a");
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        Assert.assertEquals(0, response.getErrors().size());
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java:890]: should the regulation 29 field be asserted here?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java [hunk 3]
```diff
@@ -865,4 +865,164 @@ public void givenSupportGroupRefusedScenario_Regulation29FieldIsNotPopulatedAndP
...
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java:969]: should the regulation 29 field be asserted here?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java [hunk 4]
```diff
@@ -865,4 +865,164 @@ public void givenSupportGroupRefusedScenario_Regulation29FieldIsNotPopulatedAndP
...
+        sscsCaseData.setEsaWriteFinalDecisionSchedule3ActivitiesQuestion(Arrays.asList("schedule3MobilisingUnaided"));
+
+        sscsCaseData.setWcaAppeal("Yes");
+        sscsCaseData.setWriteFinalDecisionAllowedOrRefused("allowed");
+
+        sscsCaseData.setWriteFinalDecisionGenerateNotice("yes");
+
+        sscsCaseData.setEsaWriteFinalDecisionPhysicalDisabilitiesQuestion(Arrays.asList("mobilisingUnaided"));
+
+        // 0 points - high, which means regulation 29 is not applicable
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java:964]: I'm not sure I understand this comment. What is `0 points  - high`?

**Review by ryandac — APPROVED** [review]
looks okay, minor comments and questions

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionService.java (hunk 1 above)
Comment (michaellavelle) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionPreviewDecisionService.java]: Thanks Ryan - have now replaced with a constant

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java (hunk 2 above)
Comment (michaellavelle) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java:894]: As discussed, I've ammended the comment to make this clearer

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java (hunk 4 above)
Comment (michaellavelle) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionAboutToSubmitHandlerTest.java]: As discussed, I've ammended the comment to make this clearer

**Comment by michaellavelle** [comment]
Superseded by https://github.com/hmcts/sscs-tribunals-case-api/pull/1530

**Pull request closed without merging**